chore: improve verify cookie options

StephaneRob opened the pull request:
Try to fix #668.

No breaking changes, only deprecate `Guardian.Plug.VerifyCookie` in favor of a `:verify_cookie` on `Guardian.Plug.VerifySession` and `Guardian.Plug.VerifyHeader` plugs

Hanspagh commented:
This looks great. I will have a full look later today

Hanspagh commented on test/guardian/plug/verify_header_test.exs:
```diff
@@ -172,4 +172,96 @@ defmodule Guardian.Plug.VerifyHeaderTest do
+        :get
+        |> conn("/")
+        |> put_req_header("authorization", ctx.token)
+        |> VerifyHeader.call(module: ctx.impl, verify_cookie: [module: ctx.impl])
```
To avoid all the `module: ctx.impl` madness you can use `|> Pipeline.put_module(ctx.impl)` then we can also test the option true `to verify_cookie`, what do you think?

StephaneRob replied:
I will add a test case for `refresh_from_cookie: true` (new `verify_cookie` :)).

StephaneRob commented:
Hi @Hanspagh!
I updated Readme and module doc plugs. 
I agree, the name is not the best for this options. I've also changed it to `:refresh_from_cookie`.

Hanspagh commented:
Apart from the above, I don't have anything else @ueberauth/developers ?

StephaneRob commented:
@Hanspagh,  I added a test with `refresh_from_cookie: true` and clean tests by using ` Pipeline.put_module` and ` Pipeline.put_error_handler`

Hanspagh reviewed (approved):
Looks good. No more comments from me. 👍
